### ISSUE TITLE
[Question] What's the most efficient way to count blobs under a folder?

### ISSUE THREAD

**Issue by GZTimeWalker** [robinrodricks/FluentStorage#91]
What's the most efficient way to count under a folder?

Using Count after `ListAsync` doesn't seem to be a very elegant approach, as it reads a lot of metadata from the file - even if I only need a single count number.

**Comment by robinrodricks** [comment]
We could introduce a new count method. Suggest the signature. Also suggest which underlying AWS/Azure .NET SDK API it would use.

Which cloud are you using?

**Comment by GZTimeWalker** [comment]
Not much, I knew FluentStorage just a day before...

And about the storage, I have only used minio and aws s3.

**Comment by GZTimeWalker** [comment]
For signature, I think this is fine:

```cs
/// <summary>
/// Returns the count of available blobs
/// </summary>
/// <param name="cancellationToken"></param>
/// <returns>Count of blobs</returns>
Task<int> CountAsync(string fullPath, CancellationToken cancellationToken = default);
```

For most cases, count is no need to do recursively. As expected, is should only access one folder blob.

NOTE: maybe an option for count Regular Files or Folders or both of them.

**Comment by GZTimeWalker** [comment]
cc @thompson-tomo, I see you post a lot of issues, would you consider this suggestion as well?

**Comment by thompson-tomo** [comment]
Tbh I don't see much use for a count method as you wouldn't get the benefit you think as almost all of the time the count would just be calling tolist().count anyway. The key thing for me is look at what the providers offer and go from there. For instance maybe a getFileNames() method which can provide the count.

**Comment by GZTimeWalker** [comment]
Hi, I still want a way to count only the folder meta through the filesystem......
When there are thousands of files in a folder, it's a bad behavior to read all of them in order to count them, it's slow.

**Comment by thompson-tomo** [comment]
I agree but we are limited to what the storage provider exposes by their sdk. Hence I would explore what they provide and then come up with a proposal.

**Comment by robinrodricks** [comment]
Good idea. Are you looking at cloud buckets or FTP/disk for this feature?
